writer.go

```diff
@@ -210,6 +210,29 @@ func (w *Writer) ReadRowsFrom(rows RowReader) (written int64, err error) {
 // The returned value will be nil if no schema has yet been configured on w.
 func (w *Writer) Schema() *Schema { return w.schema }
 
+// KeyValueMetadata sets a key/value pair in the Parquet file metadata.
+//
+// Keys are assumed to be unique, if the same key is repeated multiple times the
+// last value is retained. While the parquet format does not require unique keys,
+// this design decision was made to optimize for the most common use case where
+// applications leverage this extension mechanism to associate single values to
+// keys. This may create incompatibilities with other parquet libraries, or may
+// cause some key/value pairs to be lost when open parquet files written with
+// repeated keys. We can revisit this decision if it ever becomes a blocker.
+func (w *Writer) KeyValueMetadata(key, value string) {
```

Comment:
How about renaming to `Set*` to better communicate that the previous value will be overwritten?

We could also imagine having an `Add*` method for cases where the application needs to associated multiple values to a key.
```suggestion
func (w *Writer) SetKeyValueMetadata(key, value string) {
```

Reply:
Updated in the latest commit.